src/main/java/cat/udl/eps/softarch/demo/handler/DonationEventHandler.java

```diff
@@ -17,15 +22,31 @@ public class DonationEventHandler {
     final Logger logger = LoggerFactory.getLogger(User.class);
     final UserRepository userRepository;
     final DonationRepository donationRepository;
+    final RequestRepository requestRepository;
+    final TakeRepository takeRepository;
 
-    public DonationEventHandler(UserRepository userRepository, DonationRepository donationRepository) {
+    public DonationEventHandler(UserRepository userRepository, DonationRepository donationRepository, RequestRepository requestRepository, TakeRepository takeRepository) {
         this.userRepository = userRepository;
         this.donationRepository = donationRepository;
+        this.takeRepository = takeRepository;
+        this.requestRepository = requestRepository;
     }
 
     @HandleBeforeCreate
     public void handleDonationPreCreate(Donation donation) {
         logger.info("Before creating: {}", donation.toString());
+        Request request = donation.getTarget();
+        if (request != null){
+            Take take = new Take();
+            take.setAmount(request.getAmount());
+            take.setWeight(request.getWeight());
+            take.setLocation(request.getLocation());
+            take.setTakePropagator(request.getPropagator());
+            donation.setTakenBy(take);
+            takeRepository.save(take);
```

Comment:
Please, add at least one backend test to check if this implementation is working as expected

Reply:
Done.